packages/data-stores/src/agents-manager/actions.ts

```diff
@@ -0,0 +1,130 @@
+import { default as apiFetchPromise } from '@wordpress/api-fetch';
+import { Location } from 'history';
+import { default as wpcomRequestPromise, canAccessWpcomApis } from 'wpcom-proxy-request';
+import { GeneratorReturnType } from '../mapped-types';
+import type { APIFetchOptions } from '../shared-types';
+
+/**
+ * Partial state object for saving agents manager preferences
+ */
+type AgentsManagerState = {
+	isOpen?: boolean;
+	isDocked?: boolean;
+	sessionId?: string | null;
+	routerHistory?: null; // Only used to clear history
+};
+
+/**
+ * Save the agents manager state to the remote user preferences.
+ * @param state - Partial state object with the properties to save
+ * @yields {Object} Action to set router history if closing
+ */
+export function* saveAgentsManagerState( state: AgentsManagerState ) {
+	const saveState: Record< string, boolean | string | null > = {};
+
+	if ( typeof state.isOpen === 'boolean' ) {
+		saveState.agents_manager_open = state.isOpen;
+
+		//TODO: Figure out the trigger to reset the chat on agents manager
+		// Help center resets when closing. But the agents manager might be different.
+		if ( ! state.isOpen ) {
+			saveState.agents_manager_router_history = null;
+			yield setAgentsManagerRouterHistory( undefined );
+		}
+	}
+
+	if ( typeof state.isDocked === 'boolean' ) {
+		saveState.agents_manager_docked = state.isDocked;
+	}
+
+	if ( state.sessionId !== undefined ) {
+		saveState.agents_manager_session_id = state.sessionId;
+	}
+
+	// Only make API call if there's something to save
+	if ( Object.keys( saveState ).length === 0 ) {
+		return;
+	}
+
+	if ( canAccessWpcomApis() ) {
+		// Use the promise version to do that action without waiting for the result.
+		wpcomRequestPromise( {
+			path: '/me/preferences',
+			apiNamespace: 'wpcom/v2',
+			method: 'PUT',
+			body: { calypso_preferences: saveState },
+		} ).catch( () => {} );
+	} else {
+		// Use the promise version to do that action without waiting for the result.
+		apiFetchPromise( {
+			global: true,
+			path: '/help-center/open-state',
```

Comment:
After discussing with @jom, we realized we need a new endpoint here that's specific to the agents-manager. He will be working on developing the new endpoint.

Reply:
I'm calling it `/agents-manager/open-state`